ros/src/waypoint_updater/waypoint_updater.py

```diff
@@ -120,20 +158,44 @@ def pose_cb(self, msg):
 
         # store location (x, y)
         self.position = msg.pose.position
+
+        # store orientation
+        self.orientation = msg.pose.orientation
+        #self.euler = tf.transformations.euler_from_quaternion(self.orientation)
+        #self.yaw = euler[2]
+
         frame_id = msg.header.frame_id
 
         if self.base_waypoints:
 
             # get closest waypoint
             index = get_closest_waypoint_index(self.position, self.base_waypoints)
 
+            # get next waypoint
+            #index2 = get_next_waypoint_index(self.position, self.yaw, self.base_waypoints)
+
             # make list of n waypoints ahead of vehicle
             lookahead_waypoints = get_next_waypoints(self.base_waypoints, index, LOOKAHEAD_WPS)
 
             # set velocity of all waypoints
             for waypoint in lookahead_waypoints:
                 waypoint.twist.twist.linear.x = 4.47 #10mph in meters per second
 
+            # fit a polynomial path
+            coefficients = fit_polynomial(lookahead_waypoints, 2)
+
+            # calculate radius curvature
+            curvature = []
+            for waypoint in lookahead_waypoints:
+                x = waypoint.pose.pose.position.x
+                radius = calculateRCurve(coefficients, x)
+                curvature.append(radius)
+
+            # set angular velocity of all waypoints
+            for waypoint in lookahead_waypoints:
+                for i in range(LOOKAHEAD_WPS):
```

Comment:
The way you do this, we may have fewer points than lookahead_wps and get list index errors
please use 

```
for waypoint in lookahead_waypoints:
                waypoint.twist.twist.linear.x = 4.47 #10mph in meters per second
                w_x = waypoint.pose.pose.position.x
                curvature = calculateRCurve(coefficients, w_x)
                if curvature > threshold:
                     waypoint.twist.twist.angular.z = waypoint.twist.twist.linear.x/curvature 
               else:
                    waypoint.twist.twist.angular.z = 0
``
to do it in one go without the danger of list index